objects/69ead2/contained/DUESETUP/script.lua

```diff
@@ -0,0 +1,643 @@
+    if card then
+        local plus_color = impendTable[guid].energy[3] == 1 and {1,1,0,1} or {1,1,1,1}
+        local plus_ind = getButtonIndex(card,"g3Plus")
+        card.editButton({
+            index = plus_ind,
+            color = plus_color,
+        })
+        local minus_color = impendTable[guid].energy[3] == -1 and {1,1,0,1} or {1,1,1,1}
+        local minus_ind = getButtonIndex(card,"g3Minus")
+        card.editButton({
+            index = minus_ind,
+            color = minus_color,
+        })
+    end
+end
+
+function g3CardsSelected()
+    local count = 0
+    for _,tbl in pairs(impendTable) do
+        if tbl.energy[3] ~= 0 then
+            count = count + 1
+        end
+    end
+    return count
+end
+
+function g3Plus(card)
+    local selected = g3CardsSelected()
+    local guid = card.guid
+    if impendTable[guid].energy[3] == 1 then
+        impendTable[guid].energy[3] = 0
+    elseif impendTable[guid].energy[3] == -1 or selected < 2 then
+        impendTable[guid].energy[3] = 1
+    end
+    updateEnergyDisplay(guid)
+    updateG3Buttons(guid)
+    updateSave()
+end
+
+function g3Minus(card)
+    local selected = g3CardsSelected()
+    local guid = card.guid
+    if impendTable[guid].energy[3] == -1 then
+        impendTable[guid].energy[3] = 0
+    elseif impendTable[guid].energy[3] == 1 or selected < 2 then
+        impendTable[guid].energy[3] = -1
+    end
+    updateEnergyDisplay(guid)
+    updateG3Buttons(guid)
+    updateSave()
+end
+
+function createAllCardButtons()
+    createAllG3Buttons()
+    for guid,_ in pairs(impendTable) do
+        local card = getObjectFromGUID(guid)
+        if card then
+            Wait.time(function()
+                createCardButtons(card)
+                updateEnergyDisplay(guid)
+                if g3Selected and impendTable[guid].turnSelected ~= Global.getVar("turn") then
+                    createG3Buttons(card)
+                    updateG3Buttons(guid)
+                end
+            end, 0.01)
+        end
+    end
+    updateSave()
+end
+
+------------------
+-- other functions
+------------------
+
+function getButtonIndex(obj,str) -- needed for cards that already have buttons so can be edited normally
+    if obj.getButtons() then
+        for _,tbl in pairs(obj.getButtons()) do
+            if tbl.click_function == str then
+                return tbl.index
+            end
+        end
+    end
+    return nil
+end
+
+function timePasses()
+    local dances = Global.call("getSpirit", {name = spiritName})
+    for _,tbl in pairs(impendTable) do
+        local var = tbl.energy[1] + tbl.energy[2] + tbl.energy[3]
+        if var >= tbl.maxEnergy and tbl.turnSelected ~= Global.getVar("turn") - 1 then
+            clearCardData(tbl.guid)
+        else
+            tbl.energy[1] = var
+            tbl.energy[2] = 0
+            tbl.energy[3] = 0
+            updateEnergyDisplay(tbl.guid)
+        end
+    end
+    g3Selected = true
```

Comment:
shouldn't this be setting `g3Selected` to false?

Reply:
This intentionally sets it to true as it calls another function on the next line which toggles it to false. This was done to reuse the function that was already being used for when you press the button on G3 (toggle on/off) so another function didn't have to be written which would be identical with only one line changed. If this isn't preferable then I can easily just create such a function, or I can leave a comment on that line if clarification is all that is needed.